integration-tests/suites/udp_networkflow.go

```diff
@@ -0,0 +1,293 @@
+			LocalAddress:   "",
+			RemoteAddress:  fmt.Sprintf("%s:%d", servers[i].ip, servers[i].port),
+			Role:           "ROLE_CLIENT",
+			SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+			CloseTimestamp: types.NilTimestamp,
+		}
+	}
+
+	// We give a big period here to ensure the syscall happens just once
+	// Due to an implementation restriction, the total number of messages
+	// sent must be less than 32.
+	client := s.runClient(config.ContainerStartConfig{
+		Name:       UDP_CLIENT,
+		Image:      image,
+		Command:    newClientCmd("sendmmsg", "300", "8", servers...),
+		Entrypoint: []string{"udp-client"},
+	})
+	log.Info("Client: %s\n", client.String())
+
+	for _, server := range servers {
+		serverConnection := types.NetworkInfo{
+			LocalAddress:   fmt.Sprintf(":%d", server.port),
+			RemoteAddress:  client.ip,
+			Role:           "ROLE_SERVER",
+			SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+			CloseTimestamp: types.NilTimestamp,
+		}
+		s.Sensor().ExpectConnections(s.T(), server.id, 5*time.Second, serverConnection)
+	}
+	s.Sensor().ExpectConnections(s.T(), client.id, 5*time.Second, clientConnections...)
+}
+
+func (s *UdpNetworkFlow) TestMultipleSources() {
+	image := config.Images().QaImageByKey("qa-udp")
+	port := uint16(9100)
+
+	server := s.runServer(config.ContainerStartConfig{
+		Name:    UDP_SERVER,
+		Image:   image,
+		Command: newServerCmd("recvmmsg", port),
+	}, port)
+	log.Info("Server: %s\n", server.String())
+
+	clients := make([]containerData, CONTAINER_COUNT)
+	serverConnections := make([]types.NetworkInfo, CONTAINER_COUNT)
+	for i := 0; i < CONTAINER_COUNT; i++ {
+		name := fmt.Sprintf("%s-%d", UDP_CLIENT, i)
+		clients[i] = s.runClient(config.ContainerStartConfig{
+			Name:       name,
+			Image:      image,
+			Command:    newClientCmd("sendto", "300", "", server),
+			Entrypoint: []string{"udp-client"},
+		})
+		log.Info("Client: %s\n", clients[i].String())
+
+		// Load the server connection collector has to send for this client.
+		serverConnections[i] = types.NetworkInfo{
+			LocalAddress:   fmt.Sprintf(":%d", server.port),
+			RemoteAddress:  clients[i].ip,
+			Role:           "ROLE_SERVER",
+			SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+			CloseTimestamp: types.NilTimestamp,
+		}
+	}
+
+	clientConnection := types.NetworkInfo{
+		LocalAddress:   "",
+		RemoteAddress:  fmt.Sprintf("%s:%d", server.ip, server.port),
+		Role:           "ROLE_CLIENT",
+		SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+		CloseTimestamp: types.NilTimestamp,
+	}
+
+	for _, client := range clients {
+		s.Sensor().ExpectConnections(s.T(), client.id, 5*time.Second, clientConnection)
+	}
+	s.Sensor().ExpectConnections(s.T(), server.id, 5*time.Second, serverConnections...)
+}
+
+func newServerCmd(recv string, port uint16) []string {
+	return []string{
+		"--syscall", recv,
+		"--port", strconv.FormatUint(uint64(port), 10),
+	}
+}
+
+func (s *UdpNetworkFlow) runServer(cfg config.ContainerStartConfig, port uint16) containerData {
```

Comment:
Can we turn this into a generic thing to reduce duplication between runServer/runClient?
e.g.

```go
func (s *UdpNetworkFlow) runServer(cfg config.ContainerStartConfig, port uint16) containerData {
    return s.runContainer(cfg, port)
}

func (s *UdpNetworkFlow) runClient(cfg config.ContainerStartConfig) containerData {
    return s.runContainer(cfg, 0)
}

func (s *UdpNetworkFlow) runContainer(cfg config.ContainerStartConfig, port uint16) containerData {
    // ...
}
```